client/src/js/services/SupplierService.js

```diff
@@ -0,0 +1,41 @@
+angular.module('bhima.services')
+.service('SupplierService', SupplierService);
+
+SupplierService.$inject = [ '$http', 'util' ];
+
+function SupplierService($http, util) {
+  var service = this;
+
+  service.create = create;
+  service.read = read;
+  service.update = update;
+  service.filter = filter; 
+
+
+  function create(supplier) {
+    return $http.post('/suppliers', supplier)
+    .then(util.unwrapHttpResponse);
+  }
+
+  function read(uuid, params) {
+     var url = (uuid) ? '/suppliers/' + uuid : '/suppliers';
+     return $http.get(url, { params : params })
+     .then(util.unwrapHttpResponse);
+  }
+
+  function filter(params) {
+    var url = '/suppliers/filter';
```

Comment:
A better name for this function would be `search()`. It is called search elsewhere in the application.

It would also be good to set the default limit here.  Something like:
```js
params.limit = params.limit || 20;
```